nncf/pruning/export_helpers.py

```diff
@@ -0,0 +1,394 @@
+        input_masks = get_input_masks(nx_node, nx_graph)
+
+        nx_node['input_masks'] = input_masks
+        nx_node['output_mask'] = None
+        nx_node['accept_pruned_input'] = False
+
+
+@PRUNING_OPERATOR_METATYPES.register('stop_propagation_ops')
+class StopMaskForwardOps(DefaultMetaOp):
+    subtypes = [MeanMetatype, MaxMetatype, MinMetatype, LinearMetatype, MatMulMetatype]
+
+    def mask_propagation(self, model: NNCFNetwork, nx_node, graph: NNCFGraph, nx_graph: nx.DiGraph):
+        input_masks = get_input_masks(nx_node, nx_graph)
+
+        nx_node['input_masks'] = input_masks
+        nx_node['output_mask'] = None
+        nx_node['accept_pruned_input'] = False
+
+
+class ModelPruner:
+    def __init__(self, model: NNCFNetwork, graph: NNCFGraph, nx_graph: nx.DiGraph):
+        self.model = model
+        self.graph = graph
+        self.nx_graph = nx_graph
+
+    def node_propagate_can_prune_attr(self, nx_node_key):
+        nx_node = self.nx_graph.nodes[nx_node_key]
+        is_conv = self.graph.node_type_fn(nx_node) in Convolution().get_all_op_aliases()
+        return not is_conv or (is_conv and nx_node['is_depthwise'])
+
+    def node_accept_different_inputs(self, nx_node):
+        """
+        Return whether nx_node accept pruned and not pruned inputs as inputs at the same time.
+        """
+        node_type = self.graph.node_type_fn(nx_node)
+        return node_type in Concat.get_all_op_aliases()
+
+    @staticmethod
+    def get_class_by_type_name(type_name):
+        """
+        Return class of metaop that corresponds to type_name type.
+        """
+        cls = PRUNING_OPERATOR_METATYPES.get_operator_metatype_by_op_name(type_name)
+        if cls is None:
+            raise RuntimeError("Class {} is not found".format(type_name))
+            # cls = StopMaskForwardOps
+        return cls
+
+    def mask_propagation(self):
+        """
+        Mask propagation in graph:
+        to propagate masks run method mask_propagation (of metaop of current node) on all nodes in topological order.
+        """
+        sorted_nodes = [self.nx_graph.nodes[node_name] for node_name in nx.topological_sort(self.nx_graph)]
+        for node in sorted_nodes:
+            node_type = self.graph.node_type_fn(node)
+            cls = self.get_class_by_type_name(node_type)()
+            cls.mask_propagation(self.model, node, self.graph, self.nx_graph)
+        nncf_logger.info('Finished mask propagation in graph')
+
+    def propagate_can_prune_attr_up(self):
+        """
+        Propagating can_prune attribute in reversed topological order.
+        This attribute depends on accept_pruned_input and can_prune attributes of output nodes.
+        Node can_prune is True if all outputs accept_pruned_input is True and all outputs
+        (except convs because conv can be pruned by input and output independently).
+        """
+        for node_name in self.nx_graph.nodes:
+            self.nx_graph.nodes[node_name]['can_prune'] = True
```

Comment:
I would use constant variable for this string as well, because it's used in multiple places and there's a chance to introduce  a bug on refactoring